### PR TITLE
domain: fix goroutine leak in auto analyze worker

### PR DESCRIPTION
**Pull request by lamxTyler**
PTAL @coocood @tiancaiamao

### CODE REVIEW

File: domain/domain.go [hunk 1]
```diff
@@ -612,14 +613,21 @@ func (do *Domain) autoAnalyzeWorker(lease time.Duration) {
...
+	defer analyzeTicker.Stop()
 	for {
-		if statsOwner.IsOwner() {
-			err := statsHandle.HandleAutoAnalyze(do.InfoSchema())
-			if err != nil {
-				log.Error("[stats] auto analyze fail:", errors.ErrorStack(err))
+		select {
+		case <-analyzeTicker.C:
+			if statsOwner.IsOwner() {
+				err := statsHandle.HandleAutoAnalyze(do.InfoSchema())
```
Comment (coocood) [inline domain/domain.go]: `HandleAutoAnalyze` takes very long time.
We should be about to quit in the middle of `HandleAutoAnalyze` if `domain` exit.

Comment (lamxTyler) [reply domain/domain.go]: The `HandleAutoAnalyze` will analyze at most one table/index in one time, but currently, we cannot exit in the middle of an `analyze`.

Comment (tiancaiamao) [reply domain/domain.go]: You can pass domain.exit channel, it would be closed when domain is closed.

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**